### ISSUE TITLE
Clearing authentication after defined length of time

### ISSUE THREAD

**Issue by troddick** [USGS-WiM/SiGLDMS#63]
Security issue: After weeks of not using the DMS (or having it open) and multiple restarts, the DMS kept me logged in when I accessed the main URL. While EPA’s user feedback liked the ability to stay logged in for an entire day, it probably shouldn’t remember our credentials for weeks on end. (It, of course, logs me out of if I click “Log Out”, but not if I close the browser window or application.)

**Comment by troddick** [comment]
https://github.com/HackedByChinese/ng-idle

ng-Idle seems the best choice for this. Can detect if $window is being closed, but can not do anything about it because the window is closed.. Setting idle length is best option. If user hasn't done anything (mouse click, request, etc) after so many minutes, $cookies will be cleared and user will be pointed to login page. Will look into this more..

**Comment by troddick** [comment]
@jlbruce  How long of an idle should I allow? You mention they liked being able to stay logged in for an entire day. Should I allow 8 hrs of idle, then logoff? T

**Comment by jlbruce** [comment]
@troddick Out of curiosity, is the idle specifically looking for interaction with the DMS, or just any computer or browser usage, even if it's outside of the DMS? Will it give a warning that they're about to be logged off?

**Comment by jlbruce** [comment]
Either way, 8 hours should be fine.

**Comment by troddick** [comment]
@jlbruce   .. So, I am not sure if this is fixed or not. I found a solution, but in order to implement it, I had to update my libraries (versions to a newer one). I then applied the fix by adding an expiration date to the session cookie. I added a 1 hr expiration. Pushed it to the sigldev server. Tested it, and when I reopened the browser after logging in and closing it, I was asked to log in. But it should of kept me logged in for an hour. So, then I removed the fix (expiration) and tried again. Same result. So, now I'm wondering if just updating my versions of libraries did the fix. .. or I'm really missing something here. Can you please clear your cache, log in, close your browser and reopen it to see if you are still logged in?
Thanks!

**Comment by jlbruce** [comment]
Yep, still logged in. Sorry. Tested in Firefox and Chrome.

(And FYI, it let me log into the lampadmin account from both Firefox and
Chrome at the same time. I'm hoping it's allowing it because I'm at the
exact same IP, and not an issue with not preventing multiple users from
sharing the same account?)
- \+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +
  _Jennifer Bruce_
  Cartographer + Communications Specialist, SiGL Project Lead
  USGS Wisconsin Water Science Center
  608-821-3906
  jlbruce@usgs.gov
  _wi.water.usgs.gov http://wi.water.usgs.gov_

**Comment by troddick** [comment]
So for some reason it wasn't clearing the creds for me like it did last night when I opened, logged in, closed and reopened.. I was still logged in. So, I re-added the fix I applied yesterday (added 1 hr expiration to the auth cookie). Logged in, closed the browser, and waited an hour. I just opened the  browser and I wasn't logged in anymore! Yah!. I will extend it to 8 hours now. 
As for the logging in on multiple browsers, I'd like to table this until I talk with Jeremy a little more on it. I'll look at other issues and get back to you on this.
